gpytorch/likelihoods/gaussian_likelihood.py

```diff
@@ -17,22 +17,24 @@ def __init__(self, noise_covar):
         super().__init__()
         self.noise_covar = noise_covar
 
-    def forward(self, input, *params):
-        if not isinstance(input, MultivariateNormal):
-            raise ValueError("Gaussian likelihoods require a MultivariateNormal input")
-        mean, covar = input.mean, input.lazy_covariance_matrix
+    def _shaped_noise_covar(self, base_shape, *params):
         if len(params) > 0:
             # we can infer the shape from the params
             shape = None
         else:
             # here shape[:-1] is the batch shape requested, and shape[-1] is `n`, the number of points
-            shape = mean.shape
-        noise_covar = self.noise_covar(*params, shape=shape)
-        full_covar = covar + noise_covar
-        return input.__class__(mean, full_covar)
+            shape = base_shape
+        return self.noise_covar(*params, shape=shape)
 
-    def variational_log_probability(self, input, target):
-        raise NotImplementedError
+    def forward(self, function_samples, *params, **kwargs):
```

Comment:
this would be a good place to start with type hints to make clear that function_samples are tensors and function_dist below is an instance of a gpytorch MultivariateNormal

Reply:
We need to land the rtd yaml file before adding type hints